main/src/main/java/cgeo/geocaching/storage/DataStore.java

```diff
@@ -3560,6 +3560,21 @@ public static void saveIndividualRoute(final List<RouteItem> routeItems) {
         });
     }
 
+    public static void removeFirstMatchingIdFromIndividualRoute(final String id) {
+        withAccessLock(() -> {
+            init();
+            database.beginTransaction();
+            try {
+                database.delete(dbTableRoute, "precedence = (SELECT precedence FROM " + dbTableRoute + " WHERE id = ? OR id LIKE ? ORDER BY precedence LIMIT 1)", new String[] { id, id + "-%" });
```

Comment:
On removing a cache from an IndividualRoute this would also remove all waypoints of this cache from the IndividualRoute (resp. the first one of them), as their generated id start with the same prefix (geocode). To get the identifier for the element to be removed have a look at `RouteItem.buildIdentifier()`